### PR TITLE
Class moving refactor

### PR DESCRIPTION
**Pull request by Paola-Xaq**
Refactor of Classes:
- KnightMove
- PawnMove
- LinealMove

Fixing Castling class, an error delete all the class.

### CODE REVIEW

File: app/src/main/java/src/Castling.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            if(!isHaveMoved(Chessboard.board[piece.getInitPosY()][xPosQ])) {
+                checkEmpty = 0;
+                for(int i = 1; i <= 3; i++) {
+                    if(!isSpaceEmpty(Chessboard.board[piece.getInitPosY()][xPosQ + i])) {
+                        break;
+                    }
+                    checkEmpty++;
+                }
+                if(checkEmpty == 3)
+                    movePositions.add(new Position(xPosQ + 2, piece.getInitPosY()));
```
Comment (SaulCM) [inline app/src/main/java/src/Castling.java:43]: Magic number

File: app/src/main/java/src/Castling.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    checkEmpty++;
+                }
+                if(checkEmpty == 2)
+                    movePositions.add(new Position(xPosK - 1, piece.getInitPosY()));
+            }
+
+            //Castling Queen
+            if(!isHaveMoved(Chessboard.board[piece.getInitPosY()][xPosQ])) {
+                checkEmpty = 0;
+                for(int i = 1; i <= 3; i++) {
```
Comment (SaulCM) [inline app/src/main/java/src/Castling.java:36]: Magic number 3

File: app/src/main/java/src/Castling.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                        break;
+                    }
+                    checkEmpty++;
+                }
+                if(checkEmpty == 2)
+                    movePositions.add(new Position(xPosK - 1, piece.getInitPosY()));
+            }
+
+            //Castling Queen
+            if(!isHaveMoved(Chessboard.board[piece.getInitPosY()][xPosQ])) {
```
Comment (SaulCM) [inline app/src/main/java/src/Castling.java:34]: Maybe refactor castling validations to another method and get boundaries as parameters.

File: app/src/main/java/src/LinedMove.java [hunk 4]
```diff
@@ -1,79 +1,85 @@
...
-                break;
-            }
+        List<Position> movePositions = new ArrayList<>();
+        int xPos = piece.getInitPosX();
+        int yPos = piece.getInitPosY();
+
+        for( int i = xPos + 1; i < Chessboard.DIMENSION; i++) {
+            if(isSpaceEmpty(Chessboard.board[yPos][i])) {
+                movePositions.add(new Position(i, yPos));
+            }else if(isSpaceWithEnemy(Chessboard.board[yPos][i], piece.getColorWhite())) {
```
Comment (SaulCM) [inline app/src/main/java/src/LinedMove.java:18]: Care with spaces, leave a space before the }_else and after the conditional if_()

**Review by JorgeCaceres-Jala — APPROVED** [review]

**Review by GustavoZHA — APPROVED** [review]

**Pull request merged**